sed/loader/flash/loader.py

```diff
@@ -786,11 +731,29 @@ def parquet_handler(
                 detector,
             )
 
-            # Read all parquet files using dask and concatenate into one dataframe after filling
-            dataframe = self.fill_na(
-                [dd.read_parquet(file) for file in parquet_filenames],
+            # Read all parquet files into one dataframe using dask
+            dataframe = dd.read_parquet(parquet_filenames)
+
+            # Channels to fill NaN values
+            channels: List[str] = self.get_channels_by_format(["per_pulse", "per_train"])
+
+            # Define a custom function to forward fill specified columns
+            def forward_fill_partition(df):
+                df[channels] = df[channels].ffill()
+                return df
+
+            # calculate the number of rows in each partition and choose least
+            nrows = dataframe.map_partitions(len)
+            max_part_size = min(nrows)
+
+            # Use map_overlap to apply forward_fill_partition
+            dataframe = dataframe.map_overlap(
+                forward_fill_partition,
+                before=max_part_size + 1,
```

Comment:
the +1 would make this fail, as described in the comment before

Reply:
sorry i didn't quite get what you meant in your previous comment. 
isn't this what you pushed in your previous commit? or are you referring to the PR description

Reply:
I am working on this currently, and will push a solution with tests soon